Add getPathData and setPathData methods (#38730)
mdn/content: files/en-us/web/api/svgpathelement/setpathdata/index.md

---
title: "SVGPathElement: setPathData() method"
short-title: setPathData()
slug: Web/API/SVGPathElement/setPathData
page-type: web-api-instance-method
browser-compat: api.SVGPathElement.setPathData
---

{{APIRef("SVG")}}

The **`SVGPathElement.setPathData()`** method sets the sequence of path segments as the new path data.

## Syntax

```js-nolint
setPathData()
```

### Return value

> The sequence of path segments.

Reviewer: No it isn't.

> 

Reviewer: Can you please also provide a live example?

---
title: "SVGPathElement: setPathData() method"
short-title: setPathData()
slug: Web/API/SVGPathElement/setPathData
page-type: web-api-instance-method
browser-compat: api.SVGPathElement.setPathData
---

{{APIRef("SVG")}}

The **`SVGPathElement.setPathData()`** method sets the sequence of path segments as the new path data.

## Syntax

```js-nolint
setPathData(pathData)
```

### Parameters

- `pathData`

>   - : An array of path segments.

Reviewer: What does each path segment look like? Same format doc probably as for the other method.

Reviewer: I'm just going to do it so we can get this in

```suggestion
  - : An array of path segments.
  
    Each path segment is an object with the following properties:

    - `type`
      - : A [path commands](/en-US/docs/Web/SVG/Reference/Attribute/d#path_commands).
        If [`options.normalize`](/en-US/docs/Web/API/SVGPathElement/getPathData#normalize) is true this will be one of the the absolute commands: `'M'`, `'L'`, `'C'` and `'Z'`.
    - `values`
      - : An array or value containing the parameters for the corresponding command.
```

### Return value

None ({{jsxref('undefined')}}).

## Example

> Consider the following `<path>` element, drawing a square:

Reviewer: ```suggestion

### Set path data

Consider the following `<path>` element, drawing a square:
```

---
title: "SVGPathElement: setPathData() method"
short-title: setPathData()
slug: Web/API/SVGPathElement/setPathData
page-type: web-api-instance-method
browser-compat: api.SVGPathElement.setPathData
---

{{APIRef("SVG")}}

The **`SVGPathElement.setPathData()`** method sets the sequence of path segments as the new path data.

## Syntax

```js-nolint
setPathData(pathData)
```

### Parameters

- `pathData`

  - : An array of path segments.

### Return value

None ({{jsxref('undefined')}}).

## Example

### Set path data

Consider the following `<path>` element, drawing a square:

```xml
<svg xmlns="http://www.w3.org/2000/svg" width="64" height="64">
  <path d="M0,0 h64 v64 h-64 z" />
</svg>
```

> The {{domxref("SVGPathElement/getPathData", "getPathData()")}} method will return normalized path data using [absolute commands](/en-US/docs/Web/SVG/Reference/Attribute/d#path_commands). Setting this data back to the `<path>` element using the `setPathData()` method will result in the different set of path commands in the DOM:

Reviewer: ```suggestion
The code below uses {{domxref("SVGPathElement/getPathData", "getPathData()")}} method to return the normalized path data as [absolute commands](/en-US/docs/Web/SVG/Reference/Attribute/d#path_commands).
Setting the returned data back to the `<path>` element using the `setPathData()` method will result in the different set of path commands in the DOM:
```
